babel.config.js

```diff
@@ -7,9 +7,11 @@ const isCalypsoClient = process.env.CALYPSO_CLIENT === 'true';
 const modules = isCalypsoClient ? false : 'commonjs'; // only calypso should keep es6 modules
 const codeSplit = require( './server/config' ).isEnabled( 'code-splitting' );
 
-const targets = isCalypsoClient
-	? { browsers: [ 'last 2 versions', 'Safari >= 10', 'iOS >= 10', 'ie >= 11' ] }
-	: { node: 'current' };
+const targets =
+	isCalypsoClient || 'true' === process.env.TARGET_BROWSER
```

Comment:
I think we should avoid anything like defaulting to Calypso Client being true, as Calypso is a bit of a special environment and we apply some transforms there that we don't necessarily want elsewhere.

Reply:
I don't love this patch, but apparently it does the job :smile:

I think eventually we should consider setting `BABEL_ENV` to something. Then we'll have 3D structure of env variables:
- `CALYPSO_ENV`: one of development, wpcalypso, horizon, staging, production
- `BABEL_ENV`: one of server, client-calypso, client-sdk? Affects only Babel transpilation
- `NODE_ENV`: a lot of tools look at this, namely webpack for the production vs development distinction

Reply:
Looking into other project, I came across `BABEL_ENV` in use in create-react-app:

https://github.com/facebook/create-react-app/blob/30ee52cf3b2cbb6ac70999c02b1196bcaba8d4ca/packages/react-scripts/scripts/build.js#L11-L13

Are you concerned about `*_ENV` variables acquiring different meanings?

Reply:
To be clear, I'm not especially attached to `TARGET_BROWSER` as a name.

Reply:
`BABEL_ENV` being set to `server` etc. could simplify the `babel.config.js` file. Instead of `isCalypsoClient` conditionals, we could just declare an `env` field:
```js
env: {
  server: {
    presets: [ [ '@babel/env', { targets: { node: 'current' } } ] ]
  }
}
```
We already do that for `test`. But can we really invent our own values or are tools already counting on `production`, `development` and `test`? I don't know.

Reply:
> `BABEL_ENV` being set to `server` etc. could simplify the `babel.config.js` file. Instead of `isCalypsoClient` conditionals, we could just declare an `env` field:
> 
> ```js
> env: {
>   server: {
>     presets: [ [ '@babel/env', { targets: { node: 'current' } } ] ]
>   }
> }
> ```

This is a nice idea and I think we should put it on a backlog. With Jetpack code freeze coming up, we need to get a fix in place so that blocks can be built correctly for browsers as we look at shipping a new bundle. I'll plan to move ahead with this less-involved approach for now. I'm happy to open an issue with this idea we can address when deadlines aren't looming.

> We already do that for `test`. But can we really invent our own values or are tools already counting on `production`, `development` and `test`? I don't know.

I tried to use the GitHub-wide search to get an idea of how prolific this is and whether it may have significance across projects like `NODE_ENV`. I don't believe that's the case and there isn't cause for concern.

Reply:
> This is a nice idea and I think we should put it on a backlog.

Totally agreed and I didn't mean to implement it right now in this PR. Just thinking out loud about the future of our Babel and webpack configs.

> I think we should avoid anything like defaulting to Calypso Client being true, as Calypso is a bit of a special environment and we apply some transforms there that we don't necessarily want elsewhere.

To share another idea for the (hopefully near) future: "The Calypso" shouldn't be a special environment forever. The only thing that makes it special right now seems to be presence of code splitting. That affects the `<AsyncLoad>` transform, splitting the moment.js locales etc. If we correctly configure all that with a `shouldCodeSplit` flag, the concept of "The Calypso" environment should disappear. It will be just another SDK build configured as "for browser with code splitting on".

Reply:
Just for the record– In addition to code splitting, there's another transform that may or may not make sense outside of Calypso:

https://github.com/Automattic/wp-calypso/blob/422b4014f26edbc6102c766be7668a73e4cad263/babel.config.js#L51
https://github.com/Automattic/wp-calypso/blob/422b4014f26edbc6102c766be7668a73e4cad263/inline-imports.js#L4-L5